File: src/main/java/com/sfh/pokeRogueBot/model/browser/pokemonjson/Stats.java
package com.sfh.pokeRogueBot.model.browser.pokemonjson;

import com.sfh.pokeRogueBot.model.enums.Stat;
import lombok.Data;

@Data
public class Stats {

    private Integer hp;

    private Integer attack;
    private Integer defense;

    private Integer specialAttack;
    private Integer specialDefense;

    private Integer speed;

}

File: src/main/kotlin/com/sfh/pokeRogueBot/model/browser/pokemonjson/Stats.kt
package com.sfh.pokeRogueBot.model.browser.pokemonjson

data class Stats(
    var hp: Int,
    var attack: Int,
    var defense: Int,
    var specialAttack: Int,
    var specialDefense: Int,
    var speed: Int
) {
    companion object {
        fun createDefault(): Stats {
            return Stats(
                20,
                5,
                5,
                5,
                5,
                5
            )
        }
    }

    fun getBaseTotal(): Int {
        return hp + attack + defense + specialAttack + specialDefense + speed
    }
}